src/models/router/ActionRoutes.php

```diff
@@ -87,32 +87,36 @@ protected function validateControllerClass($controllerClass)
 
     /**
      * Returns all available actions of the specified controller.
-     * @param Controller $controller the controller instance
+     * @param \ReflectionClass $controller reflection of the controller
      * @return array all available action IDs with optional action class name (for external actions).
-     * @throws \ReflectionException
      */
     protected function getActions($controller)
     {
         $actions = [];
 
-        $externalActions = $controller->actions();
-        foreach ($externalActions as $id => $externalAction) {
-            if (is_array($externalAction) && array_key_exists('class', $externalAction)) {
-                $actions[$id] = $externalAction['class'];
-            } elseif (is_string($externalAction)) {
-                $actions[$id] = $externalAction;
-            }
+        try {
+            // avoid potential problems with __constructor() and init()
```

Comment:
It is, as always. But it's less possible since by the definition it should be only static array of actions configurations in `actions()`. I've decided to choose lesser of two evils here.